packages/graphql/src/translate/translate-read-new.ts

```diff
@@ -0,0 +1,201 @@
+        where: { varName, node },
+    });
+    if (whereAuth[0]) {
+        whereStrs.push(whereAuth[0]);
+        cypherParams = { ...cypherParams, ...whereAuth[1] };
+    }
+
+    if (whereStrs.length) {
+        whereStr = `WHERE ${whereStrs.join(" AND ")}`;
+    }
+
+    const allowAndParams = createAuthAndParams({
+        operation: "READ",
+        entity: node,
+        context,
+        allow: {
+            parentNode: node,
+            varName,
+        },
+    });
+    if (allowAndParams[0]) {
+        cypherParams = { ...cypherParams, ...allowAndParams[1] };
+        authStr = `CALL apoc.util.validate(NOT(${allowAndParams[0]}), "${AUTH_FORBIDDEN_ERROR}", [0])`;
+    }
+
+    if (optionsInput) {
+        if (hasOffset) {
+            offsetStr = `SKIP $${varName}_offset`;
+            cypherParams[`${varName}_offset`] = optionsInput.offset;
+        }
+
+        if (hasLimit) {
+            limitStr = `LIMIT $${varName}_limit`;
+            cypherParams[`${varName}_limit`] = optionsInput.limit;
+        }
+
+        if (optionsInput.sort && optionsInput.sort.length) {
+            if (hasLimit) {
+                const sortFields = optionsInput.sort.map((sort) => Object.keys(sort)[0]);
+                const orderByProjection = createProjectionAndParams({
+                    node,
+                    context,
+                    fieldsByTypeName: {
+                        [node.name]: Object.fromEntries(
+                            Object.entries(fieldsByTypeName[node.name]).filter(([key]) => sortFields.includes(key))
+                        ),
+                    },
+                    varName,
+                });
+                [limitOrderProjStr] = orderByProjection;
+                cypherParams = { ...cypherParams, ...orderByProjection[1] };
+            }
+
+            const sortArr = optionsInput.sort.reduce((res: string[], sort: GraphQLSortArg) => {
+                return [
+                    ...res,
+                    ...Object.entries(sort).map(([field, direction]) => {
+                        return `${hasLimit ? limitOrderVarName : varName}.${field} ${direction}`;
+                    }),
+                ];
+            }, []);
+
+            sortStr = `ORDER BY ${sortArr.join(", ")}`;
+        }
+    }
+
+    let cypher: string[] = [];
+
+    if (!hasLimit) {
+        cypher = [
+            matchStr,
+            whereStr,
+            authStr,
+            ...(sortStr ? [`WITH ${varName}`, sortStr] : []),
+            ...(projAuth ? [`WITH ${varName}`, projAuth] : []),
+            ...connectionStrs,
+            `RETURN ${varName} ${projStr} as ${varName}`,
+            offsetStr,
+        ];
+    } else {
+        cypher = [
+            "CALL {",
+            matchStr,
+            whereStr,
+            authStr,
+            ...(sortStr ? [`WITH ${varName}, ${varName} ${limitOrderProjStr} AS ${limitOrderVarName}`, sortStr] : []),
+            ...(projAuth ? [`WITH ${varName}`, projAuth] : []),
```

Comment:
I'm not 100% sure what projAuth is;  but this will be evaluated for every match pre-limit, though it doesn't seem to be participating in the sorting/limiting process itself, so perhaps it could be moved out of the call black to be evaluated only for the nodes that are going to be returned?